snuba/subscriptions/worker.py

```diff
@@ -47,3 +50,24 @@ def flush_batch(self, batch: Sequence[Sequence[SubscriptionResult]]) -> None:
         for results in batch:
             for result in results:
                 self.__producer.produce(self.__topic, result).result()
+
+
+class SubscriptionResultCodec(Codec[KafkaPayload, SubscriptionResult]):
+    def encode(self, value: SubscriptionResult) -> KafkaPayload:
+        subscription_id = str(value.task.task.identifier)
+        return KafkaPayload(
+            str(subscription_id).encode("utf-8"),
```

Comment:
```suggestion
            subscription_id.encode("utf-8"),
```

This is already a str due to the coerce on line 57.